container-search/src/main/java/com/yahoo/prelude/fastsearch/FastHit.java

```diff
@@ -322,14 +322,14 @@ public String toString() {
     @Override
     public int hashCode() {
         if (getId() == null) {
-            throw new IllegalStateException("This hit must have a 'uri' field, and this fild must be filled through " +
+            throw new IllegalStateException("This hit must have a 'uri' field, and this field must be filled through " +
                                             "Execution.fill(Result)) before hashCode() is accessed.");
         } else {
             return super.hashCode();
         }
     }
 
-    private static StringBuilder asHexString(StringBuilder sb, GlobalId gid) {
+    private void appendAsHex(GlobalId gid, StringBuilder sb) {
```

Comment:
This can and should still be static, right?

Reply:
Can: Yes, Should: Not imho

Reply:
When should a helper that could be static not be declared as such?

Reply:
When it's not meant to be used in a static context.

Reply:
Do as you want. A good SO thread: https://stackoverflow.com/questions/538870/should-private-helper-methods-be-static-if-they-can-be-static